equinox-shipping/Domain/Shipment.fs

```diff
@@ -0,0 +1,58 @@
+﻿module Shipment
+
+type ContainerId = string
+
+type Shipment = { created: bool; container: ContainerId option }
+
+module Events =
+
+    let [<Literal>] CategoryId = "Shipment"
+    let streamName clientId = FsCodec.StreamName.create CategoryId clientId
+
+    type Event =
+        | ShipmentCreated
+        | ShipmentAssigned of {| containerId : string |}
+        | ShipmentUnassigned
+        interface TypeShape.UnionContract.IUnionContract
+
+    let codec = FsCodec.NewtonsoftJson.Codec.Create<Event>()
+
+module Fold =
+
+    type State = Shipment
+
+    let initial: State = { created = false; container = None }
+
+    let evolve (state: State) (event: Events.Event): State =
+        match event with
+        | Events.ShipmentCreated          -> { state with created = true }
+        | Events.ShipmentAssigned   event -> { state with container = Some event.containerId }
+        | Events.ShipmentUnassigned       -> { state with container = None }
+
+
+    let fold: State -> Events.Event seq -> State =
+        Seq.fold evolve
+
+type Command =
+    | Create
```

Comment:
what is the goal of a) having one of these b) showing it in this sample

Reply:
I'll remove it from the sample.

Reply:
Removed thanks!